packages/flutter_test/lib/src/matchers.dart

```diff
@@ -352,6 +385,185 @@ class _HasOneLineDescription extends Matcher {
   Description describe(Description description) => description.add('one line description');
 }
 
+class _NormalizeHashCodesEquals extends Matcher {
+  _NormalizeHashCodesEquals(String v) : _value = _normalize(v);
+
+  final String _value;
+
+  static final Object _mismatchedValueKey = new Object();
+
+  static String _normalize(String s) {
+    return s.replaceAll(new RegExp(r'#[0-9a-f]{5}'), '#000');
+  }
+
+  @override
+  bool matches(dynamic object, Map<dynamic, dynamic> matchState) {
+    final String description = _normalize(object);
+    if (_value != description) {
+      matchState[_mismatchedValueKey] = description;
+      return false;
+    }
+    return true;
+  }
+
+  @override
+  Description describe(Description description) {
+    return description.add('multi line description equals $_value');
+  }
+
+  @override
+  Description describeMismatch(
+    dynamic item,
+    Description mismatchDescription,
+    Map<dynamic, dynamic> matchState,
+    bool verbose
+  ) {
+    if (matchState.containsKey(_mismatchedValueKey)) {
+      final String actualValue = matchState[_mismatchedValueKey];
+      // Leading whitespace is added so that lines in the multi-line
+      // description returned by addDescriptionOf are all indented equally
+      // which makes the output easier to read for this case.
+      return mismatchDescription
+          .add('expected normalized value\n  ')
+          .addDescriptionOf(_value)
+          .add('\nbut got\n  ')
+          .addDescriptionOf(actualValue);
+    }
+    return mismatchDescription;
+  }
+}
+
+/// Returns `true` if [c] represents a whitespace code unit.
+bool _isWhitespace(int c) => (c <= 0x000D && c >= 0x0009) || c == 0x0020;
+
+/// Returns `true` if [c] represents a vertical line unicode line art code unit.
+///
+/// See [https://en.wikipedia.org/wiki/Box-drawing_character]. This method only
+/// specifies vertical line art code units currently used by Flutter line art.
+/// There are other line art characters that technically also represent vertical
+/// lines.
+bool _isVerticalLine(int c) {
+  return c == 0x2502 || c == 0x2503 || c == 0x2551 || c == 0x254e;
+}
+
+/// Returns whether a [str] is all vertical tree connector characters.
+///
+/// Example vertical tree connector characters: `│ ║ ╎`.
+/// The last line of a text tree contains only vertical tree connector
+/// characters indicates a poorly formatted tree.
+bool _isAllTreeConnectorCharacters(String str) {
```

Comment:
If done sparingly and for the main or only argument of a short and very generic method, or in math expressions, single-character variable names are probably ok.

As a general rule I would avoid them, though, unless they actively help readability. So for example, they often help readability when you're expressing physics equations, and `i` is probably not less readable than `index` when looping over something.